docs/monix-catnap.md

```diff
@@ -0,0 +1,18 @@
+# monix-catnap
+
+[![Maven Central](https://img.shields.io/maven-central/v/com.avast/sst-monix-catnap_2.12)](https://repo1.maven.org/maven2/com/avast/sst-monix-catnap_2.12/)
+
+`libraryDependencies += "com.avast" %% "sst-monix-catnap" % "<VERSION>"`
+
+This module provides interop between Scala Server Toolkit and [monix-catnap](https://monix.io/docs/3x/#monix-catnap) library.
+
+## Circuit Breaker
+
+You can use `CircuitBreakerModule` to instantiate and configure a circuit breaker and you can implement `CircuitBreakerMetrics` to get
+monitoring of the circuit breaker. There is an implementation for Micrometer in `sst-monix-catnap-micrometer` module.
+
+All of this is tied with http4s HTTP client in the `sst-http4s-client-monix-catnap` module so in practice you want to use
+`Http4sClientCircuitBreakerModule` which wraps any `Client[F]` with a `CircuitBreaker` and adds logging/metrics. However
+the most important feature of the enriched circuit breaker is that any HTTP failure (response `5xx`) is converted to an exception internally
```

Comment:
i think this should be configurable - by that i mean it should be configurable what triggers a failure in circuit breaker - for example `408 Request Timeout` or `429 Too Many Requests` seem like they should trigger CB

Reply:
configurable is definitely the way to go. that being said, `5xx` is a good default IMHO

Reply:
sure - another thing to consider is that i think CB should be somehow enabled per IP address and not http client but that's probably a lot more work

Reply:
I'll make it configurable, good point.

I am not sure it's right to do it per IP but even if it was I don't think it's doable in the current implementation. Right now it works on the level of `Request => Response` and the `Request` contains URL with domain (not the actual target IP). We would have to do this inside the HTTP client. We can consider this somehow in the future but I don't want to block this PR with this right now.

Reply:
`408` is actually a client failure (at least according to wiki) so I did not use that but I added `429` as default.